host/bcm283x/dma.go

```diff
@@ -758,20 +760,30 @@ func startPWMbyDMA(p *Pin, rng, data uint32) (*dmaChannel, *videocore.Mem, error
 		drvGPIO.gpioBaseAddr + 0x1C + 4*uint32(p.number/32), // set
 	}
 	// High
-	if err := cb[0].initBlock(physBit, dest[1], data*4, false, true, false, false, dmaPWM); err != nil {
-		_ = buf.Close()
-		return nil, nil, err
+	ptr := physBuf
+	for i := uint32(0); i < data; i++ {
```

Comment:
My only worry is that this could be fairly large in egregious cases. Can you calculate what's the worst case?

For example if PWM(DutyHalf, 1Hz) is called, rng would be huge.

Reply:
With:
drvDMA.pwmDMAFreq = 200 * physic.KiloHertz
freq = 1 * physic.Hertz

We have:
rng := uint64(drvDMA.pwmDMAFreq / freq)
rng = 200 000

And we allocate:
rng*controlBlockSize + 4
200000 * 32 + 4 = 6400004
Rounded to 4KB:
6402048 bytes => 6252 kiB or around 6MiB

So, to progress on that question:
* Is it too much or not (I have frankly no idea)? Maybe I can return an error if it's too much.
* Do you know what kind of device would require such low-frequency PWM signal? I know servos are in the range 50Hz - 200Hz, more than 100kHz for neopixel LEDs, but there's probably plenty of other devices I don't know.
* I like the cleanness of the solution of using only 2 control blocks, but I don't explain why it's unstable with multiple DMA channels. I unfortunately couldn't find much documentation on this. However, shouldn't the source buffer as large as this size?

Reply:
It's a significant amount, as it's allocated from the GPU memory. The default amount available is 64MiB as per https://www.raspberrypi.org/documentation/configuration/config-txt/memory.md.

1Hz is borderline egregious; 50Hz (like for servos) is 125Kb, which is tolerable. Can you just add a comment about the trade off being made here?

Reply:
I'd say as documentation for startPWMbyDMA. Technically the CL is good as is, so I can commit and the comment can be added as a follow up, since you already waited a fair amount to get this in.